modules/examiner/jsx/editExaminer.js

```diff
@@ -0,0 +1,355 @@
+import Panel from 'Panel';
+import Loader from 'Loader';
+
+/**
+ * Examiner Module
+ *
+ * Edit examiner page to the module that allows changes to the examiner's
+ * certification status for enabled instruments
+ *
+ * Renders the Edit Examiner page consisting of FormElement
+ * and Panel components.
+ *
+ * @author Zaliqa Rosli
+ * @version 1.0.0
+ */
+class EditExaminer extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      data: {},
+      formData: {},
+      statusOptions: {
+        null: "N/A",
+        notCertified: "Not Certified",
+        inTraining: "In Training",
+        certified: "Certified"
+      }
+    };
+
+    /**
+     * Bind component instance to custom methods
+     */
+    this.fetchData = this.fetchData.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.editForm = this.editForm.bind(this);
+    this.changeLog = this.changeLog.bind(this);  
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   */
+  fetchData() {
+    const self = this;
+    let formData = {};
+    $.ajax({
+      url: this.props.dataURL,
+      type: 'GET',
+      dataType: 'json',
+      data: {
+        identifier: this.props.examinerID
+      },
+      success: data => {
+        const instruments = data.instruments;
+        for (let instrumentID in instruments) {
+          if (instruments.hasOwnProperty(instrumentID)) {
+            formData[instrumentID] = {
+              status: data.instruments[instrumentID].pass,
+              date: data.instruments[instrumentID].date_cert,
+              comments: data.instruments[instrumentID].comment
+            };
+          }
+        }
+        self.setState({
+          data: data,
+          formData: formData,
+          isLoaded: true
+        });
+      },
+      error: error => {
+        console.error(error);
+      }
+    });
+  }
+
+  /**
+   * Store the value of the element in this.state.formData
+   *
+   * @param {string} formElement  name of the form element
+   * @param {string} value        value of the form element
+   */
+  setFormData(formElement, value) {
+    let formData = this.state.formData;
+    // Form elements have format name="instrumentID_fieldName"
+    // We want to extract that information to be used as keys
+    // in the nested formData array
+    let splitNameOfElement = formElement.split("_");
+    let instrumentID = splitNameOfElement[0];
+    let fieldName = splitNameOfElement[1];
```

Comment:
I'm pretty sure variables on lines 94-96 can be declared as `const`.

EDIT: After looking up `let` and `const` some more we really should be using `const` in more places - even for arrays that have values within them changed. I don't want to nit-pick about what should and shouldn't be a `const`, but consider reviewing the code and changing all variables that don't get reassigned to `const`